src/meta_memcache/base/cache_pool.py

```diff
@@ -393,7 +388,7 @@ def delta_and_get(
         result = self.meta_arithmetic(
             key=key, flags=flags, int_flags=int_flags, token_flags=token_flags
         )
-        if isinstance(result, Value):
+        if isinstance(result, Value) and result.value is not None:
             return int(result.value)
```

Comment:
I think the proper behavior if the value is `None` should be `int()` throwing an error. Returning `None` is intended for misses. If `meta_arithmetic` returns a `Value(None)` is the same as if it returns `Value('not_a_number')`, both should not happen, point to other issues or problems.

I think the ugliness of this interface is my fault. Perhaps we should make meta_arithmetic return a new special `IntValue` that always contains an int? It will reduce the complexity on this layer. And any value that is not int is converted to failure, so this layer doesn't need to handle this craziness. :) We can create a `WriteIntResponse` for the union as well.

About the check itself, perhaps something like:
```
if isinstance(result, Value):
    if isinstance(result.value, str) and result.value.isnumeric()
        return int(result.value)
    else:
        raise MemcacheError(f"Unexpected value from meta arithmetic command: {result.value}")
```